packages/core/src/asciidoctor-core-api.js

```diff
@@ -1035,7 +1035,8 @@ AbstractNode.prototype.isRole = function (expectedValue) {
  * @memberof AbstractNode
  */
 AbstractNode.prototype.getRole = function () {
-  return this.$role()
+  const role = this.$role()
+  return Opal.nil ? undefined : role
```

Comment:
I think it should be:

```suggestion
  return role === Opal.nil ? undefined : role
```

a test will confirm that :wink:

Reply:
Have you any idea where to put a test for it...I mean to test in test for types seems a bit weird...but would be do for now...maybe to restructure/split in general test.ts would be a good idea for more structure. So what are your thoughts?

Reply:
> Have you any idea where to put a test for it...I mean to test in test for types seems a bit weird...but would be do for now...maybe to restructure/split in general test.ts would be a good idea for more structure. So what are your thoughts?

Yes, at some point, we should `tests.ts`, the file is starting to get quite long but we should do that in a dedicated pull request.

The idea is to add a test in `tests.ts` to make sure that the type definition is correct.

You can add a test in `packages/core/spec/share/asciidoctor-spec.js`.
The idea is to load a document and call `getRole` on a block that does not have a role:

_untested-code.js_
```js
const doc = asciidoctor.load(`= Title

== Section without role
`)
const sectionWithoutRole = doc.getBlocks()[0]
expect(sectionWithoutRole.getRole()).to.be.undefined()
```

All tests in "share/asciidoctor-spec.js" will be executed against all supported runtime, currently:
- Node
- Browser
- GraalVM